src/Agent/NewRelic/Agent/Core/DataTransport/Client/NRHttpClient.cs

```diff
@@ -22,7 +23,10 @@ public class NRHttpClient : HttpClientBase
         public NRHttpClient(IWebProxy proxy, IConfiguration configuration) : base(proxy)
         {
             _configuration = configuration;
-            _httpClientWrapper = new HttpClientWrapper(new HttpClient(new HttpClientHandler { Proxy = proxy }, true));
+
+            // set the default timeout to "infinite", but specify the configured collector timeout as the actual timeout for SendAsync() calls
```

Comment:
Do we need to set this timeout to infinite? It seems like it would be dangerous to do if some bug prevented the request level timeout from being applied.

Reply:
The default timeout has to be greater than any timeout we'd want to specify. But since we always specify a timeout for the cancellation token passed to the `.SendAsync()` call (in HttpClientWrapper), it's not an issue.